components/Form/ScoreInput.tsx

```diff
@@ -0,0 +1,150 @@
+import { css, StyleSheet } from "aphrodite";
+import { ReactElement, useEffect, useMemo, useState } from "react";
+import colors from "~/config/themes/colors";
+
+type Props = {
+    value?: number;
+    onSelect?: Function;  
+    readOnly?: Boolean;
+    overrideBarStyle?: any;
+    scoreInputStyleOverride?: any;
+    withText ?: Boolean;
+};
+
+const MAX_SCORE = 10;
+const MIN_SCORE = 1;
+
+function ScoreInput({
+  value = MIN_SCORE,
+  onSelect,
+  readOnly,
+  overrideBarStyle = null,
+  scoreInputStyleOverride = null,
+  withText = true,
+}: Props): ReactElement {
+
+  const [selectedValue, setSelectedValue] = useState<number>(value);
+  const [hoveredValue, setHoveredValue] = useState<number>(0);
+
+  useEffect(() => {
+    if (value !== selectedValue) {
+      setSelectedValue(value);
+    }
+  }, [value])
+
+  const handleSelect = (value) => {
+    setSelectedValue(value);
+    onSelect && onSelect(value);
+  }
+
+  const buildBarInput = () => {
```

Comment:
was there a specific reason why this was a functional? 
If not, let's just relocate this to be inside of useMemo. Or maybe making this its own component outside of  `ScoreInput` is better (I understand that keep it inside is convenient for variable access)

Reply:
sounds reasonable. I think I had a reason earlier on.